Show date (or weekday) of game

AleXSR700 opened the pull request:
Please, see https://github.com/pigskin/kodi-gamepass/issues/452

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update settings.xml

AleXSR700 committed:
Update settings.xml

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update strings.po

AleXSR700 committed:
Update settings.xml

AleXSR700 committed:
Update strings.po

AleXSR700 committed:
Update settings.xml

AleXSR700 committed:
Update settings.xml

AleXSR700 committed:
Update settings.xml

AleXSR700 committed:
Update settings.xml

AleXSR700 committed:
Update settings.xml

AleXSR700 committed:
Update settings.xml

AleXSR700 commented:
Unfortunately github shows each of my test commits which makes it look a bit weird. I have tested and seems to all work as intended :)

Feel free to pimp ;)

jm-duke commented:
Thanks for your contribution! I'm a bit busy right now, so please give me a few days for reviewing the PR :)

Cheers,
Duke

jm-duke commented on resources/language/resource.language.en_gb/strings.po:
```diff
@@ -145,6 +145,10 @@ msgctxt "#30047"
+msgctxt "#30047"
+msgid "PRESEASON WEEK {0}"
+msgstr ""
+
```
This is a duplicate now, I assume you copied it by mistake?

AleXSR700 replied:
Yes, did not notice that. Should not be duplicated.

jm-duke commented on resources/settings.xml:
```diff
@@ -66,12 +66,23 @@
+          <dependencies>
+            <dependency type="enable" operator="!is" setting="hide_game_length">false</dependency>
+          </dependencies>
+        </setting>
```
Hm, I don't really like cyclic dependencies, i.e. two items depending on each other. Is there a reason why you added the dependencies?

AleXSR700 replied:
The dependency is needed because if you do not hide the game length, the second line is used for game length and can no longer be used for datetime. 
So either you show game length in line 2 or datetime. If you wanted both, the layout would need adjusting. 
Alternativ was to create an either or dependency.

jm-duke replied:
Ah, I see. That makes sense indeed, I didn't have the layout in mind when checking the code. Thanks for the explanation

jm-duke reviewed (changes requested):
All in all I can see that this is working as expected.

Can you please comment on and/or change the code I commented on?
Once that's done, I'll test it on my Kodi

AleXSR700 commented on default.py:
```diff
@@ -280,16 +287,16 @@ def display_weeks_games(self):
+                    else:
+                        game_info = game['phase']
+                        if addon.getSetting('hide_game_length') == 'true' and game['phase'] != 'FINAL':
+                            game_info = 'FINAL' + '\n' + '(' + datetime_obj.strftime(datetime_format) +')'
```
Not sure where you see the issue.  To be honest, this is just moving of your original code. So not mine (I just moved it forward to be able to access it sooner). But it works as intended, at least for the 24 h clock I am using.

AleXSR700 replied:
I think you are correct. Wierd that I did not encounter this during testing  🤔

AleXSR700 commented:
I commented and will make the corrections tomorrow so you can test :)

AleXSR700 committed:
Update default.py

AleXSR700 committed:
Update strings.po

AleXSR700 commented:
@jm-duke 
I made the corrections except for the dependency (as explained in my response). Of course, please feel free to change it if you have a cleaner approach.
We can also adjust the layout more, but I find it quite nice as is so I did not want to mess with it more than necessary.

Of course I cannot test if all changes will work correctly when we actually have upcoming and ongoing games. But I will test that as soon as the season kicks off *.*

AleXSR700 committed:
Update strings.po

jm-duke commented:
I'll test the suggested changes on my debug kodi and will merge, if everything works out.

And yes, we should re-test once the pre-season is available to see if the change messes anything up then :)

jm-duke committed:
Remove whitespaces

jm-duke committed:
Updating default and condition for new setting

When I enabled the proposed changes, both settings were disabled and
I wasn't able to change either. So I had to adjust the logic as follows:
  - By default, both settings are false to keep the current behaviour as
    default
  - display_datetime is only displayed, if hide_game_length == true
  - hide_game_length is only displayed, if display_datetime == false

jm-duke committed:
Grouping dependent settings

Swapped display_current_week and display_datetime, because display_datetime
is linked with hide_game_length, so they should be next to one another

jm-duke commented:
With the proposed changes, both settings (hide_game_length and display_datetime) were disabled and I was not able to change either. I had to change the dependency logic a bit to make it work, looks good on my end now.
Can you retest on your kodi?

jm-duke committed:
Add msgid 30052 to other language files

AleXSR700 commented:
Thanks for re-testing. Due to the iterations my default state got changed.
I will test tomorrow, yes.
'!is false' and 'is true' are essentially the same in a boolean :)

AleXSR700 commented:
Performed a re-test including deleting settings and re-installing.
Everything seems to be working :)
